src/actors/realtime_update_actors.rs

```diff
@@ -28,15 +28,48 @@ pub struct RealTimeReloader {
 }
 
 // TODO: make this async
-fn fetch_gtfs_rt(url: &str) -> Result<GtfsRT, Error> {
+fn fetch_gtfs_rt(url: &str) -> Option<GtfsRT> {
     info!("fetching a gtfs_rt");
     let gtfs_rt = reqwest::get(url)
         .and_then(|resp| resp.error_for_status())
-        .map_err(|e| format_err!("Unable to fetch the gtfs RT {}", e))?
-        .bytes()
-        .collect::<Result<Vec<u8>, _>>()?;
+        .map_err(|e| format_err!("Unable to fetch GTFS: {}", e))
+        .and_then(|resp| {
+            resp.bytes()
+                .collect::<Result<Vec<u8>, _>>()
+                .map_err(|e| format_err!("Unable to decode protobuf {}", e))
+        });
+
+    match gtfs_rt {
+        Ok(gtfs_rt) => Some(GtfsRT {
+            data: gtfs_rt,
+            datetime: chrono::Utc::now(),
+        }),
+        Err(e) => {
+            log::warn!("Unable to fetch GTFS: {}", e);
+            None
+        }
+    }
+}
+
+fn aggregate_rts(feed_messages: &[transit_realtime::FeedMessage]) -> Result<GtfsRT, Error> {
```

Comment:
have you checked that the resulting gtfs_rt (that we expose in pb and json via the api) are valid ?

Reply:
yep, it's working

Reply:
for such an important feature, I think it's important to have integration tests.

this way we can check that /gtfs_rt is a valid resulting gtfs_rt and that /stop_monitoring works as expected